### PR TITLE
feat: add watch client

### PR DESCRIPTION
**Pull request by LingKa28**
Base on #3 

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
  Add new feature.

* what changes does this pull request make?
   * add watch client & test samples

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
  No.

### CODE REVIEW

File: client/watch.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+// Creates a new maintenance client
+func newWatchClient(innerWatchClient xlineapi.WatchClient) watchClient {
+	return watchClient{watchClient: innerWatchClient}
+}
+
+// Build inner client
+func buildWatchClientFromAddrs(addrs []string) (*xlineapi.WatchClient, error) {
+	conn, err := grpc.Dial(addrs[0], grpc.WithTransportCredentials(insecure.NewCredentials()))
```
Comment (iGxnon) [inline client/watch.go:24]: Could we reuse the `conn`?

**Review by Phoenix500526 — APPROVED** [review]
LGTM

File: tests/watch_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		watcher, err := watchClient.Watch()
+		assert.NoError(t, err)
+
+		err = watcher.Send(&xlineapi.WatchRequest{
+			RequestUnion: &xlineapi.WatchRequest_CreateRequest{
+				CreateRequest: &xlineapi.WatchCreateRequest{
+					Key: []byte("watch01"),
+				},
+			},
+		})
```
Comment (iGxnon) [inline tests/watch_test.go:34]: We can add a method to simplify the sending of this request.

Comment (LingKa28) [reply tests/watch_test.go:34]: This issue is related to client ease of use, and is planned to be optimized in a later pr.

**Review by Phoenix500526 — APPROVED** [review]

**Pull request merged**